spec/app_spec.rb

```diff
@@ -606,25 +606,99 @@ def app
 
   end
 
-  context "/<CHANNEL>/<PROJECT>/versions endpoint" do
+  context '/<CHANNEL>/<PROJECT>/versions endpoint' do
     Chef::Project::KNOWN_PROJECTS.each do |project|
-      context project do
+      context "for #{project}" do
         let(:endpoint){ "/stable/#{project}/versions" }
 
         it "exists" do
-          get endpoint
+          get(endpoint)
           expect(last_response).to be_ok
         end
 
         it "returns the correct JSON data" do
-          get endpoint
+          get(endpoint)
           expect(last_response.header['Content-Type']).to include 'application/json'
           expect(last_response.body).to match(project)
         end
       end
     end
-  end
 
+    context "for chef" do
+      let(:endpoint) { '/stable/chef/versions' }
+      let(:params) { { v: version } }
+      let(:versions_output) {
+        get(endpoint, params)
+        metadata_json = last_response.body
+        JSON.parse(metadata_json)
+      }
+
+      [ nil, 'latest', '12'].each do |version|
+        context "with version #{version.inspect}" do
+          let(:version) { version }
+          let(:expected_versions) {
+            JSON.parse(File.read(File.join(SPEC_DATA, "version_info/chef-latest.json")))
+          }
+
+          it 'returns the latest version for each platform, platform_version and architecture' do
+            versions_output.each do |p, data|
+              data.each do |pv, data|
+                data.each do |m, metadata|
+                  expect(metadata['relpath']).to be_a(String)
+                  expect(metadata['md5']).to match /^[0-9a-f]{32}$/
+                  expect(metadata['sha256']).to match /^[0-9a-f]{64}$/
+                  expect(metadata['url']).to match 'http'
```

Comment:
I skipped it because I would need to just copy paste the existing spec data. Found it redundant.